### PR TITLE
Systematics-aware skim and VBSjjlnu changes

### PR DESCRIPTION
**Pull request by valsdav**
This PR includes the changes for SMP-20-013 and some improvements for the framework:

- Systematics-aware skimming script
- Multiprocessing plotting to plot each cut in parallel
- Customization script for plotting with on-the-fly manipulation of config files 
- Some PlotFactory tweaks.

-------------------------------------------------------------
### Skim with systematics 

-  check the cut expression on all the variations, applying the appropriate branch mapping
- Take the OR of all the cuts
- Copy the trees keeping only the selected entries. 

Implemented job submission on condor. 
The cut is modified using the mapping defined in branch_mapping_cfg. 

### Plot customization script
Imagine  you have to do many plots with different graphical configurations (for example different ordering for each category), or including/excluding samples and nuisances in different cuts. Instead of changing manually the samples/cuts/nuisances script, a customization script can manipulate those files depending on a "tag" passed to mkPlot. 

For an example of the customization script: https://github.com/UniMiBAnalyses/PlotsConfigurations/blob/VBSjjlnu_v7/Configurations/VBSjjlnu/Full2018v7/conf_fit_v4.5/customize.py

### CODE REVIEW

**Comment by valsdav** [comment]
I will cross-check the updates in this branch in order to merge asap

File: ShapeAnalysis/python/PlotFactory.py [hunk 1]
```diff
@@ -1159,7 +1212,8 @@ def makePlot(self, inputFile, outputDirPlots, variables, cuts, samples, plot, nu
 
                 # If dividing by bin width, yaxis should be "<Events / [unit]>"
                 if variable["divideByBinWidth"] == 1:
-                    frameDistro.GetYaxis().SetTitle("< Events / %s >"%unit)
+                    # frameDistro.GetYaxis().SetTitle("< Events / %s >"%unit)
+                    frameDistro.GetYaxis().SetTitle("dN/d"+variable['xaxis'])
```
Comment (valsdav) [inline ShapeAnalysis/python/PlotFactory.py:1216]: from SMP-20-013 review. Maybe others prefers the <Events / > label

Comment (amassiro) [reply ShapeAnalysis/python/PlotFactory.py:1216]: Can we have this configurable with an option "1, 2, 3, ..." that will use different styles?

File: ShapeAnalysis/python/PlotFactory.py [hunk 2]
```diff
@@ -1051,10 +1103,10 @@ def makePlot(self, inputFile, outputDirPlots, variables, cuts, samples, plot, nu
               #                     if there is "histo_total" there is no need of explicit nuisances
               if len(mynuisances.keys()) != 0 or histo_total!= None:
                   if self._showIntegralLegend == 0 :
-                      tlegend.AddEntry(tgrMC, "All MC", "F")
+                      tlegend.AddEntry(tgrMC, "Syst.", "F")
                   else :
                       print " nexpected  = ", nexpected
-                      tlegend.AddEntry(tgrMC, "All MC [" + str(round(nexpected,1)) + "]", "F")
```
[the remark this answers is not in the document]
Comment (amassiro) [reply ShapeAnalysis/python/PlotFactory.py:1057]: Can we have this configurable with an option "1, 2, 3, ..." that will use different styles (keeping a default option somewhere)?

File: ShapeAnalysis/python/PlotFactory.py (hunk 1 above)
Comment (valsdav) [reply ShapeAnalysis/python/PlotFactory.py:1220]: would you suggest a name for this option?

Comment (lenzip) [reply ShapeAnalysis/python/PlotFactory.py:1220]: if this is the only holdback, maybe we can merge it as is and tweak it later

Comment (valsdav) [reply ShapeAnalysis/python/PlotFactory.py:1220]: You are right, let me revert my change.

**Comment by valsdav** [comment]
This PR can be now merged

**Pull request merged**